crates/bevy_ecs_compile_fail_tests/tests/ui/non_sync_ref.rs

```diff
@@ -0,0 +1,26 @@
+use bevy_ecs::prelude::*;
+use std::cell::Cell;
+
+#[derive(Component, Eq, PartialEq, Debug)]
+struct A(Cell<usize>);
+
+fn main() {
+    let mut world = World::default();
+    let e = world.spawn().insert(A(Cell::new(10_usize))).id();
+
+    {
+        let query = QueryState::<&A>::new(&mut world);
+    }
+
+    {
+        let value = world.get::<A>(e);
```

Comment:
We could 'send' at least one of these to another thread, to demonstrate the UB being avoided here.

Not sure how necessary that is though.